compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3276,36 +3283,61 @@ public Boolean visit(BType target, BType source) {
 
         @Override
         public Boolean visit(BArrayType target, BType source) {
+            source = getReferredType(source);
+            if (source.tag == TypeTags.INTERSECTION) {
+                source = getEffectiveTypeForIntersection(source);
+            }
```

Comment:
Is this the correct place to do this? We have to repeat it in the individual visits? Isn't there a common/single entry point where we can do this instead?

Reply:
+1. We can move this check to a common place. Will update it. Thanks.

Reply:
Moved this into `isSameOrderedType(BType source, BType target, Set<TypePair> unresolvedTypes)` method.